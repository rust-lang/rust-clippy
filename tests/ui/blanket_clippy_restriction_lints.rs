//@compile-flags: -W clippy::restriction

#![warn(clippy::blanket_clippy_restriction_lints)]

//! Test that the whole restriction group is not enabled
#![warn(clippy::restriction)]
#![deny(clippy::restriction)]
#![forbid(clippy::restriction)]

fn main() {}
