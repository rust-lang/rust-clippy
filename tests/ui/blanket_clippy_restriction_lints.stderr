error: `clippy::restriction` is not meant to be enabled as a group
   |
   = note: because of the command line `--warn clippy::restriction`
   = help: enable the restriction lints you need individually
   = note: `-D clippy::blanket-clippy-restriction-lints` implied by `-D warnings`

error: non-workspace dependency `clippy_lints`
   |
note: the lint level is defined here
  --> $DIR/blanket_clippy_restriction_lints.rs:10:11
   |
LL | #![forbid(clippy::restriction)]
   |           ^^^^^^^^^^^^^^^^^^^
   = note: `#[forbid(clippy::require_workspace_dependencies)]` implied by `#[forbid(clippy::restriction)]`

error: non-workspace dependency `rustc_tools_util`

error: non-workspace dependency `tempfile`

error: non-workspace dependency `termize`

error: `clippy::restriction` is not meant to be enabled as a group
  --> $DIR/blanket_clippy_restriction_lints.rs:6:9
   |
LL | #![warn(clippy::restriction)]
   |         ^^^^^^^^^^^^^^^^^^^
   |
   = help: enable the restriction lints you need individually

error: `clippy::restriction` is not meant to be enabled as a group
  --> $DIR/blanket_clippy_restriction_lints.rs:8:9
   |
LL | #![deny(clippy::restriction)]
   |         ^^^^^^^^^^^^^^^^^^^
   |
   = help: enable the restriction lints you need individually

error: `clippy::restriction` is not meant to be enabled as a group
  --> $DIR/blanket_clippy_restriction_lints.rs:10:11
   |
LL | #![forbid(clippy::restriction)]
   |           ^^^^^^^^^^^^^^^^^^^
   |
   = help: enable the restriction lints you need individually

error: aborting due to 8 previous errors

